feat: Consider UI5 Manifest v2 (#780)
UI5/linter: docs/Rules.md

## unsupported-api-usage

Checks whether the UI5 API is used correctly, for example, whether a formatter in a JavaScript/TypeScript binding declaration is of type `function`.

## no-outdated-manifest-version

Legacy-free UI5 requires Manifest Version 2. This test checks the `_version` property in `manifest.json`.

## no-removed-manifest-property

Manifest Version 2 lacks several properties of earlier versions, which do not match anymore to the new schema. They must be omitted.

## no-legacy-ui5-version-in-manifest

> Checks the `sap.ui5/dependencies/minUI5Version` property in `manifest.json` for legacy UI5 versions, which must not be used.

Author: Thanks! 
I have updated the documentation!

Hopefuly, it's clearer now

## unsupported-api-usage

Checks whether the UI5 API is used correctly, for example, whether a formatter in a JavaScript/TypeScript binding declaration is of type `function`.

## no-outdated-manifest-version

For legacy free UI5, required manifest version is 2. This test checks the `_version` property in `manifest.json`.

## no-removed-manifest-property

> In manifest version 2, some of the properties are removed and they do not match anymore to the new schema.

Reviewer: ```suggestion
Manifest Version 2 lacks several properties of earlier versions, which do not match anymore to the new schema. They must be omitted.
```

## no-renamed-manifest-property

> In manifest version 2, some of the properties are renamed, so the new names must be adjusted to match the schema.

Reviewer: ```suggestion
Manifest Version 2 renames several properties of earlier versions. Their names must be adjusted to match the new schema.
```

> ## no-legacy-ui5-version-in-manifest-2

Author: According to the BLI, it should always recommend usage of non legacy versions. So, it should be checked independent of the _version property. I'll update the rule ID

> Check for legacy UI5 versions in `minUI5Version` property in `manifest.json`

Reviewer: ```suggestion
Checks the `minUI5Version` property in `manifest.json` for legacy UI5 versions, which must not be used.
```

> Checks the `minUI5Version` property in `manifest.json` for legacy UI5 versions, which must not be used.

Author: Done

## no-incorrect-manifest-property-value

> Checks whether property `type` is explicitly set to `View` in manifest version 2 when `sap.ui5/routing/config/type` is not omitted or set to `View`.

Reviewer: ```suggestion
In Manifest Version 2, checks whether the `type` property is explicitly set to `View` when `sap.ui5/routing/config/type` is not omitted or set to `View`.
```
